primitives/derive/src/lib.rs

```diff
@@ -0,0 +1,15 @@
+mod socket_addr;
+
+/// Convenience macro to concisely construct a [`SocketAddr`]
```

Comment:
I'm afraid this won't resolve. You can try by running `cargo doc --no-deps`. Although we have a lot of unresolved links we shouldn't make more. And we don't want to add `iroha_primitives` dependency just for that. So I suggest this:

```suggestion
/// Convenience macro to concisely construct a `SocketAddr`
```

Reply:
Yeah, fair enough

Reply:
Hm... There is actually a way to make it resolve:

When re-exporting the macro in `iroha_primitives`, one can force the re-export to be inline and write the docs there. Then the macro will be documented in `iroha_primitives` crate with a resolved `SocketAddr` link. It doesn't seem to work for intellij-rust though, only shows up in rustdoc. Not sure if it's worth it.

```rust
/// Convenience macro to concisely construct a [`SocketAddr`]
///
/// # Examples
/// ```
/// # use iroha_primitives::addr::socket_addr2;
/// let localhost = socket_addr2!(127.0.0.1:8080);
/// let remote = socket_addr2!([2001:db8::1]:8080);
/// ```
#[doc(inline)]
pub use iroha_primitives_derive::socket_addr as socket_addr2;
```